mesonbuild/compilers/d.py

```diff
@@ -683,6 +692,17 @@ def get_linker_always_args(self) -> T.List[str]:
     def get_disable_assert_args(self) -> T.List[str]:
         return ['-frelease']
 
+# LDC uses the DMD frontend code to parse and analyse the code.
+# It then uses LLVM for the binary code generation and optimizations.
+# This function retrieves the dmd frontend version, which determines
+# the common features between LDC and DMD.
+# We need the complete version text because the match is not on first line
+# of version_output
+def find_ldc_dmd_frontend_version(version_output: str):
+    version_regex = re.search(r'DMD v(\d+\.\d+\.\d+)', version_output)
+    if version_regex is not None and len(version_regex.groups()):
+        return version_regex.groups()[0]
```

Comment:
Two things here, using `len()` in python to check if a container is empty is considered an anti-pattern, there's a well defined conversion from container to bool in logiacl operations, so just `if version_regex.groups()` will test if it's empty or not. The problem with len() is that it's O(n), but using the bool conversion will short circuit as soon as a single member is found.

In thisc as though I think we can simplify this to:
```suggestion
    if version_regex:
        return version_regex.group(1)
```
Note that group 0 is the entire match.